enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/actions/AddADBAction.java

```diff
@@ -65,80 +73,104 @@
     "AddADB=Add Oracle Autonomous DB",
     "SelectTenancy=Select Tenancy",
     "SelectCompartment=Select Compartment",
-    "SelectDatabase=Select Database"
+    "SelectDatabase=Select Compartment or Database",
+    "EnterUsername=Enter Username",
+    "EnterPassword=Enter Password"
 })
 public class AddADBAction implements ActionListener {
     private static final Logger LOGGER = Logger.getLogger(AddADBAction.class.getName());
     
+    private static final String DB = "db";
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
 
     @Override
     public void actionPerformed(ActionEvent e) {
-        List<TenancyItem> tenancies = new ArrayList<>();
-        for (OCIProfile p : OCIManager.getDefault().getConnectedProfiles()) {
-           p.getTenancy().ifPresent(tenancies::add);
-        }
-        Optional<TenancyItem> selectedTenancy = chooseOneItem(tenancies, Bundle.SelectTenancy());
-        
-        Optional<CompartmentItem> selectedCompartment = Optional.empty();
-                
-        if (!selectedTenancy.isPresent()) {
-            return;
-        }
-        
-        List<CompartmentItem> compartments = CompartmentNode.getCompartments().apply(selectedTenancy.get());
-        selectedCompartment = chooseOneItem(compartments, Bundle.SelectCompartment());
-        DatabaseItem selectedDatabase = null;
+        Map<String, Object> result = new HashMap<> ();
         
-        if (selectedCompartment.isPresent()) {
-            while(selectedDatabase == null) {
-                OCIItem item = chooseCopartmentOrDb(selectedCompartment.get());
-                if (item == null) {
-                    return;
-                }
-                if (item instanceof DatabaseItem) {
-                    selectedDatabase = (DatabaseItem) item;
-                }
-                if (item instanceof CompartmentItem) {
-                    selectedCompartment = Optional.of((CompartmentItem) item);
+        NotifyDescriptor.ComposedInput ci = new NotifyDescriptor.ComposedInput(Bundle.AddADB(), 2, new Callback() {
```

Comment:
There are at least 3 steps: database, username, password. In addition, each compartment selection should change the number of estimated steps.

Reply:
Increasing estimated number of steps for additional steps in the middle (sub-compartments)